crates/api/src/community/pending_follows/approve.rs

```diff
@@ -0,0 +1,46 @@
+use activitypub_federation::config::Data;
+use actix_web::web::Json;
+use lemmy_api_common::{
+  community::ApproveCommunityPendingFollower,
+  context::LemmyContext,
+  send_activity::{ActivityChannel, SendActivityData},
+  utils::is_mod_or_admin,
+  SuccessResponse,
+};
+use lemmy_db_schema::{
+  source::community::{CommunityFollower, CommunityFollowerForm},
+  traits::Followable,
+};
+use lemmy_db_views::structs::LocalUserView;
+use lemmy_utils::error::LemmyResult;
+
+pub async fn post_pending_follows_approve(
+  data: Json<ApproveCommunityPendingFollower>,
+  context: Data<LemmyContext>,
+  local_user_view: LocalUserView,
+) -> LemmyResult<Json<SuccessResponse>> {
+  is_mod_or_admin(
```

Comment:
I thought about whether this should be community mods only, but I think its fine to allow admins to do this also.

Whether the `list` endpoint should show everything to admins, whether they mod that community or not, is the bigger question.

Reply:
You currently need to pass the community id explicitly for list. That should really be changed to include all modded communities, with an option for admins to include all communities.

Reply:
Fixed